packages/trampoline/src/types.ts

```diff
@@ -0,0 +1,44 @@
+export type AnyGenerator = Generator<any, any, any>;
+/**
+ * A function type that represents a generator function for trampolining.
+ *
+ * @template TInitial - The type of the initial value.
+ * @template TArg - The type of the argument passed to the thunk function.
+ * Defaults to `TInitial`.
+ * @template TResult - The type of the result produced by the thunk function.
+ * Defaults to `TArg`.
+ * @param thunk - The thunk function to be used in the generator.
+ * @param initial - The initial value to start the generator.
```

Comment:
The implementation is not opinionated about the arguments of the generator function nor the way it gets the sync or async utilities.

It can be by dependency injection of sorts (including passing a thunk) but it could also be by choosing the right utility function from imported package based on the input arguments via any logic. 

I removed the opinionation from the trampoline params but wasn't exactly sure what the two params are up to here.